CMakeLists.txt

```
@@ -276,6 +276,11 @@ cmake_dependent_option(
 cmake_dependent_option(
     USE_GLOO "Use Gloo. Only available if USE_DISTRIBUTED is on." ON
     "USE_DISTRIBUTED" OFF)
+# If USE_GLOO then try to build Gloo with OpenSSL(currently available only on Linux).
+# set USE_GLOO_WITH_OPENSSL=OFF if you don't have OpenSSL or don't want to use it in Gloo
+cmake_dependent_option(
+    USE_GLOO_WITH_OPENSSL "Use Gloo with OpenSSL. Only available if USE_GLOO is on." ON
+    "USE_GLOO" OFF)
```

Can you please remove `AND LINUX` check from line 332 and move it here
```suggestion
cmake_dependent_option(
    USE_GLOO_WITH_OPENSSL "Use Gloo with OpenSSL. Only available if USE_GLOO is on." ON
    "USE_GLOO AND LINUX" OFF)
```